src/main/java/com/mongodb/hibernate/jdbc/MongoPreparedStatement.java

```diff
@@ -63,10 +64,11 @@ final class MongoPreparedStatement extends MongoStatement implements PreparedSta
 
     private final List<Consumer<BsonValue>> parameterValueSetters;
 
-    public MongoPreparedStatement(
-            MongoClient mongoClient, ClientSession clientSession, MongoConnection mongoConnection, String mql) {
+    MongoPreparedStatement(
+            MongoClient mongoClient, ClientSession clientSession, MongoConnection mongoConnection, String mql)
+            throws SQLSyntaxErrorException {
         super(mongoClient, clientSession, mongoConnection);
-        this.command = BsonDocument.parse(mql);
+        this.command = parse(mql);
```

Comment:
Here we parse `BsonDocument` the same exact way `MongoStatement` does that, which involves throwing `SQLSyntaxErrorException`.

Reply:
if we don't have a hard rule, we hope we have some flexibiliity. `parse()` is a common method name and I think giving it a context improves code reading and maintainance.

Reply:
I qualified the call in 9dbc8028f8a7b43853054e7b227494a4681e73a9.

Reply:
thanks. Another compelling example is the following common usage:
```
var map = Maps.of("key1", "value1", "key2", "value2");
```